flo-runner/src/main/java/com/spotify/flo/context/FloRunner.java

```diff
@@ -244,4 +217,51 @@ public static String randomAlphaNumeric(int count) {
     }
     return builder.toString();
   }
+
+  public static class Result<T> {
+    private final Future<T> future;
+
+    Result(Future<T> future) {
+      this.future = future;
+    }
+
+    public Future<T> future() {
```

Comment:
Why `Future` (which only allows blocking `get()`) instead of `CompletableFuture` or `CompletionStage` ?